src/Orc.Automation.Tests/ScenarioManagement/Loggers/UserDocumentationBuilderLogger.cs

```diff
@@ -0,0 +1,170 @@
+        htmlStringBuilder.AppendLine("<style>");
+        htmlStringBuilder.AppendLine("* {box-sizing: border-box;}");
+        htmlStringBuilder.AppendLine(".column {  float: left;  width: 33.33%;  padding: 10px;}");
+        htmlStringBuilder.AppendLine(".column1 {  float: left;  width: 20%;  padding: 10px;}");
+        htmlStringBuilder.AppendLine(".row:after {  content: \"\";  display: table;  clear: both;}");
+        htmlStringBuilder.AppendLine("</style>");
+        htmlStringBuilder.AppendLine("</head><body>");
+        htmlStringBuilder.AppendLine($"<h1>{scenario.Name}</h1>");
+        htmlStringBuilder.AppendLine($"<h2>{scenario.Description}</h2>");
+        htmlStringBuilder.AppendLine($"<div class=\"row\"><div class=\"column1\" style=\"background-color:#ccc;\"><h2>Step</h2></div><div class=\"column\" style=\"background-color:#ccc;\"><h2>Before</h2></div><div class=\"column\" style=\"background-color:#ccc;\"><h2>After</h2></div></div>");
+
+        for (var i = 0; i < scenario.StartedSteps.Count; i++)
+        {
+            var step = scenario.StartedSteps[i];
+            htmlStringBuilder.AppendLine($"<div class=\"row\"><div class=\"column1\"><h4>{step.Name}</h4></div><div class=\"column\"><img src=\"before_{i}.jpg\" style=\"object-fit:cover;width:100%;height:100%;border: solid 1px #CCC\"/></div><div class=\"column\"><img src=\"after_{i}.jpg\" style=\"object-fit:cover;width:100%;height:100%;border: solid 1px #CCC\"/></div></div>\r\n");
+        }
+
+        htmlStringBuilder.AppendLine("</body></html>");
+
+        var indexHtmlPath = Path.Combine(GetScenarioDirectory(scenario), "Index.html");
+        File.WriteAllText(indexHtmlPath, htmlStringBuilder.ToString());
+    }
+
+    private Bitmap? ScreenShot(Rect bounds)
+    {
+        try
+        {
+            Wait.UntilResponsive(200);
+
+            var captureBitmap = new Bitmap((int)bounds.Width, (int)bounds.Height, PixelFormat.Format32bppArgb);
+            var captureGraphics = Graphics.FromImage(captureBitmap);
+            captureGraphics.CopyFromScreen((int)bounds.X, (int)bounds.Y, 0, 0, new System.Drawing.Size((int)bounds.Width, (int)bounds.Height));
+
+            captureGraphics.Dispose();
+
+            return captureBitmap;
+        }
+        catch (Exception ex)
+        {
+            TestContext.Out.WriteLine(ex.ToString());
+        }
+
+        return null;
+    }
+
+    private string GetScenarioDirectory(AutomationScenario scenario, bool checkExistence = true)
+    {
+        var scenarioGroupDirectory = string.IsNullOrWhiteSpace(scenario.Suite)
+            ? _documentationRootDirectory
+            : Path.Combine(_documentationRootDirectory, scenario.Suite);
+
+        Directory.CreateDirectory(scenarioGroupDirectory);
+
+        var scenarioDirectory = Path.Combine(scenarioGroupDirectory, scenario.Name);
+        if (checkExistence && !Directory.Exists(scenarioDirectory))
```

Comment:
I think this existence check is unneccessary? Because if you don't check, you will still return the directory but not create it?